client/src/Security/ClientVoter.php

```diff
@@ -0,0 +1,121 @@
+<?php declare(strict_types=1);
+
+namespace App\Security;
+
+use App\Entity\Client;
+use App\Entity\Organisation;
+use App\Entity\User;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\Security;
+
+class ClientVoter extends Voter
+{
+    /** @var string */
+    const VIEW = 'view';
+
+    /** @var string */
+    const EDIT = 'edit';
+
+    /** @var string */
+    const DELETE = 'delete';
+
+    /** @var Security  */
+    private $security;
+
+    /**
+     * @param Security $security
+     */
+    public function __construct(Security $security)
+    {
+        $this->security = $security;
+    }
+
+    /**
+     * @param string $attribute
+     * @param mixed $subject
+     * @return bool
+     */
+    protected function supports($attribute, $subject)
+    {
+        return in_array($attribute, [self::VIEW, self::EDIT, self::DELETE]) && $subject instanceof Client;
+    }
+
+    /**
+     * @param string $attribute
+     * @param Client $client
+     * @param TokenInterface $token
+     * @return bool
+     */
+    protected function voteOnAttribute($attribute, $client, TokenInterface $token)
+    {
+        $user = $token->getUser();
+
+        if (!$user instanceof User) {
+            // the user must be logged in; if not, deny access
+            return false;
+        }
+
+
+        switch ($attribute) {
+            case self::VIEW:
```

Comment:
Do we need this case if it's not doing anything?

Reply:
Stacking the cases above each other like this is just saying that if the attribute is view or edit then return the result of `$this->canManage()`